mitodl/odl-video-service
# Enabled edX course ID editing for collections

#### What are the relevant tickets?
Closes #830 

#### What's this PR do?
Allows users to add/edit the `edx_course_id` value for a collection if **either** of the following are true:
- They are superusers/staff
- They are a member of a collection's admin Moira lists and have a certain permission group added to their user record

#### How should this be manually tested?
Test the superuser/staff case:
- Confirm that you can add/edit the edx Course ID value for a new/existing collection if your user has the `is_staff` and/or `is_superuser` flags set to `True`

Test the non-superuser/staff case:
- Log in a user with `is_staff` and`is_superuser` set to `False`
- Add that user is added to one of the admin Moira lists for some collection
- Visit that collection detail page in the app and click the gear icon. You **should not** see the edX Course ID field at the bottom of the pop up
- In Django admin, add the "edX Course Admin" group for your user (that group should have been created in a data migration)
- Visit the same page and open up the form
- You **should** see the edX Course ID field at the bottom of the pop up, and changing the value and submitting the form should change the value in the database

#### Screenshots (if appropriate)
![ss 2020-03-20 at 14 30 00 ](https://user-images.githubusercontent.com/14932219/77197393-7c983580-6abb-11ea-820a-cb099e1de5b6.png)
